### ISSUE TITLE
hashes: Hash types do not implement `AsRef<u8; LEN>`

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#1462]
Currently hashes implement `AsRef<[u8]>` by way of the `borrow_slice_impl` macro. We should consider adding `AsRef<[u8; LEN]>` where LEN is the length of the inner byte array.

Suggested: https://github.com/rust-bitcoin/rust-bitcoin/pull/1450#discussion_r1044846265

**Comment by Kixunil** [comment]
Oh, yes, I wanted to complain about this one.

**Comment by Srg213** [comment]
After reading the docs, just wanted to clarify - LEN is of type usize, right?

**Comment by Kixunil** [comment]
@Srg213 yes, though we use it as placeholder in the issue title, it should be appropriate integer literal.

**Comment by arturomf94** [comment]
Can this one be assigned to me? I’d like to try it out.

**Comment by Kixunil** [comment]
You have it!
